.github/workflows/slsa-builder-go.yml

```diff
@@ -179,63 +225,67 @@ jobs:
           path: ${{ steps.build-gen.outputs.go-binary-name }}
           retention-days: 5
 
+      #TODO(upload): verify how the action handles malicious names.
+      - name: Upload the provenance
+        uses: actions/upload-artifact@82c141cc518b40d92cc801eee768e7aafc9c2fa2 # v2.3.1
+        with:
+          name: ${{ steps.build-gen.outputs.provenance-unsigned }}
+          path: ${{ steps.build-gen.outputs.provenance-unsigned }}
+          retention-days: 5
+
   provenance:
+    runs-on: ubuntu-latest
+    needs: build
     permissions:
-      id-token: write
+      id-token: write # needed for keyless
       # package should be removed and done by caller.
       # packages: write
       contents: read
-
-    name: provenance
-    runs-on: ubuntu-latest
-    needs: build
-    steps:    
-      - uses: actions/checkout@v2.4.0
+    steps:
+      - name: Download provenance 2
+        uses: actions/download-artifact@v2
         with:
-          repository: 'asraa/slsa-on-github'
-      - uses: actions/setup-go@v2
-      # - name: build provenance generator
-      #   run: |
-      #     cd slsa-provenance/generator/
-      #     go build -o ../../prov ./cmd/main.go
-      #     cd
-      #     
-
-      # Download build from previous job
-      - uses: 'actions/download-artifact@v2'
+          name: "${{ env.PROVENANCE_BINARY }}"
+
+      - name: Download provenance 
+        uses: actions/download-artifact@v2
         with:
-          name: ${{ needs.build.outputs.go-binary-name }}
+          name: "${{ env.GENERATED_BINARY }}.intoto"
```

Comment:
Why not keep `${{ needs.build.outputs.go-binary-name }}`? This is computed using the user-defined config. Keeping it means binary and prov all have the same name. Is there a problem if we do that?

Reply:
Ah, no we should do that. cleaning that up.